### ISSUE TITLE
Indentation: Lambda in Enum causes error loop

### ISSUE THREAD

**Issue by pavelicii** [checkstyle/checkstyle#11103]
Check documentation: https://checkstyle.org/config_misc.html#Indentation
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac SomeEnum.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="Indentation"/>
    </module>
</module>

/var/tmp $ cat SomeEnum.java
package some.pckg;

import java.util.function.Function;

public enum SomeEnum {

    ENUM_VALUE(
            v -> v, // violation, but it is not expected
            new Object()
    );

    private final Function<Object, Object> function;
    private final Object object;

    SomeEnum(Function<Object, Object> function, Object object) {
        this.function = function;
        this.object = object;
    }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-9.2-all.jar -c config.xml SomeEnum.java
Starting audit...
[ERROR] C:\test\src\main\java\some\pckg\SomeEnum.java:8:13: 'lambda arguments' has incorrect indentation level 12, expected level should be 16. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```

---

I expect Checkstyle to pass without errors. If I try to increase indentation like [ERROR] suggests (from 12 to 16), I get the same error, but with indentation levels 16 and 20 (and so on). So the error is looped. Similar thing happened in #1141.

**Comment by yshubham8419** [comment]
```
ENUM_VALUE(v -> v,
         new Object()
 );

```
this works perfectly fine

**Comment by pavelicii** [comment]
@yshubham8419 not sure hot it relates to the described issue.

**Comment by Binita-tech** [comment]
@pavelicii, I am new to this type of issue, can u please guide me on how to proceed further?

**Comment by pbludov** [comment]
@Binita-tech indentations issues are the hardest to fix. Please peek another issue.
If, nevertheless, you want to do it, then look at the closed issues with the `indentation` tag:
https://github.com/checkstyle/checkstyle/issues?q=label%3Aindentation+is%3Aclosed

**Comment by Binita-tech** [comment]
Thank you @pbludov. I will be working on this issue.

**Comment by Binita-tech** [comment]
I am working on it @yshubham8419, You can pick up another issue.

**Comment by yshubham8419** [comment]
@Binita-tech if you haven't made much progress please let me do it.

**Comment by Binita-tech** [comment]
In which files do I need to make changes to make it happen?
I understood what we are trying to do here by reading the conversation in linked issue and PRs but as a newbie to this kind of issue, I just don't know in which files we need to make changes.
If you can help a little, I think I would be able to do it. @pbludov

**Comment by pbludov** [comment]
@Binita-tech you should add a test case to `IndentationCheckTest.java` like this [one](https://github.com/checkstyle/checkstyle/pull/10093/files#diff-c28a810048bbe1457a19635f94c854d8faa68ac13025b6a3e80713ae6ad96bda)

Put the submitted code to new input file like [this](https://github.com/checkstyle/checkstyle/pull/10093/files#diff-48584f03d2c127b8823efcdb4c4b9eb9da647d581504e6d91224b7d4147cf563)

Make sure that the test is failing. Then you should debug this test. Pay attention to the `LambdaHandler` class. Also, the bug may be in the `ClassDefHandler` class.

**Comment by Binita-tech** [comment]
Do I have to create input file by myself? 
Here submitted code means code in issue description right?
This is the first time, I will be writing test cases and I have observed by seeing other test cases that test cases are almost similar, In this case, is test case is similar to the test case that you have given above and if not what else do I need to know to write it? @pbludov

**Comment by pbludov** [comment]
> Do I have to create input file by myself?

Yes. It is always better to create new input file.

**Comment by MUzairS15** [comment]
@Binita-tech you still working on this?

**Comment by Binita-tech** [comment]
You can work on this @MUzairS15

**Comment by vivek-0509** [comment]
i am working on this
